docker/pypi/wmagent/run.sh

```diff
@@ -0,0 +1,683 @@
+            _check_mysql || { echo "ERROR: MaridDB database unreachable or not cleaned"; return $(false) ;}
+            ;;
+        oracle)
+            $oracleCred   || { echo "ERROR: No Oracle database credentials provided at $WMA_ADMIN_DIR/WMAgent.secrets"; return $(false) ;}
+            _check_oracle || { echo "ERROR: Oracle database unreachable or not cleaned"; return $(false) ;}
+        ;;
+    esac
+}
+
+check_docker_init() {
+    # A function to check all previously populated */.dockerInit files
+    # from all previous steps and compare them with the /data/.dockerBuildId
+    # if all do not match we cannot continue - we consider configuration/version
+    # mismatch between the host and the container
+
+    local DOCKER_INIT_LIST="
+        $WMA_INSTALL_DIR
+        $WMA_CONFIG_DIR
+        $WMA_CONFIG_DIR/wmagent
+        $WMA_CONFIG_DIR/mysql
+        $WMA_CONFIG_DIR/couchdb
+        $WMA_CONFIG_DIR/rucio
+        $WMA_HOSTADMIN_DIR
+        "
+    local stepMsg="Performing checks for successful Docker initialisation steps..."
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+    # touch $DOCKER_INIT_LIST
+    local dockerInitId=""
+    local dockerInitIdValues=""
+    local idValue=""
+    for initFile in $DOCKER_INIT_LIST; do
+        initFile=$initFile/.dockerInit
+        _init_valid $initFile && idValue=$(cat $initFile 2>&1) || idValue=$initFile
+        dockerInitIdValues="$dockerInitIdValues $idValue"
+    done
+    dockerInitId=$(for id in $dockerInitIdValues; do echo $id; done |sort|uniq)
+    echo "WMA_BUILD_ID: $WMA_BUILD_ID"
+    echo "dockerInitId: $dockerInitId"
+    [[ $dockerInitId == $WMA_BUILD_ID ]] && { echo "OK"; return $(true) ;} || { echo "ERROR"; return $(false) ;}
+
+}
+
+activate_agent() {
+    local stepMsg="Performing $FUNCNAME"
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+    _init_valid $WMA_CONFIG_DIR || {
+        echo "$FUNCNAME: triggered."
+        $manage activate-agent || { echo "ERROR: Failed to activate WMAgent!"; return $(false) ;}
+    }
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+init_agent() {
+    local stepMsg="Performing $FUNCNAME"
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+    _init_valid $WMA_CONFIG_DIR || {
+        echo "$FUNCNAME: triggered."
+        $manage init-agent || { echo "ERROR: Failed to initialise WMAgent databases!"; return $(false) ;}
+    }
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+agent_tweakconfig() {
+    local stepMsg="Performing $FUNCNAME"
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+    _init_valid $WMA_CONFIG_DIR || {
```

Comment:
To re-confirm, these lines below only get executed if there is no `.dockerInit` file (thus, first initialization of the agent), right?

Reply:
Absolutely